engine/src/main/java/com/sandboni/core/engine/finder/scanner/DirectoryScanner.java

```diff
@@ -35,9 +35,7 @@ public DirectoryScanner(String scannerName) {
 
         Map<String, Set<File>> locationFiles = new HashMap<>();
         Collection<Map<String, Set<File>>> locationToFilesFound = getDirectoryFinder(consumer).execute(allLocations);
-        locationToFilesFound.forEach(map -> {
-            map.forEach(locationFiles::put);
-        });
+        locationToFilesFound.forEach(map -> map.forEach(locationFiles::put));
```

Comment:
other way, using streams:
Map<String, Set<File>> locationFiles = locationToFilesFound.stream()
    .flatMap (map -> map.entrySet().stream()) // this would create a flattened stream of all the map entries
    .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue, (x,y) -> y));